frontend/include/chpl/resolution/resolution-queries.h

```diff
@@ -253,6 +253,17 @@ const types::QualifiedType& returnType(Context* context,
                                        const TypedFnSignature* sig,
                                        const PoiScope* poiScope);
 
+/**
+  Compute the types for any generic 'out' formal types after instantiation
+  of any other generic arguments.
+
+  The 'out' formal types are inferred from the body of the function.
+  The returned TypedFnSignature* will have the inferred out formal types.
+ */
+const TypedFnSignature* inferOutFormals(Context* context,
```

Comment:
If there is a return type declared and it is concrete, `returnType` will resolve and result in that type. So, it does not necessarily infer anything at all (it might resolve something instead).

In contrast, for an out formal, if the type is declared & concrete, there is nothing for inferOutFormals to do, since the TypedFnSignature passed to it will already have that information resolved.